lineapy/graph_reader/artifact_collection.py

```diff
@@ -206,148 +208,83 @@ def _sort_session_artifacts(
-            self._extract_session_module(
-                session_artifacts=session_artifacts,
-                indentation=indentation,
-            )
-            for session_artifacts in session_artifacts_sorted
-        ]
-
-        # Combine components by "type"
         module_imports = "\n".join(
-            [module["session_imports"] for module in session_modules]
+            [
+                sa.get_session_module_imports()
+                for sa in session_artifacts_sorted
+            ]
         )
-        artifact_functions = "\n".join(
-            [module["artifact_functions"] for module in session_modules]
+
+        artifact_function_definition = "\n".join(
+            list(
+                itertools.chain.from_iterable(
+                    [
+                        sa.get_session_artifact_function_definitions(
+                            indentation=indentation
+                        )
+                        for sa in session_artifacts_sorted
+                    ]
+                )
+            )
         )
+
         session_functions = "\n".join(
-            [module["session_function"] for module in session_modules]
-        )
-        module_function_body = "\n".join(
-            [module["session_calculation"] for module in session_modules]
-        )
-        module_function_return = ", ".join(
-            [module["session_function_return"] for module in session_modules]
+            [
+                sa.get_session_function(indentation=indentation)
+                for sa in session_artifacts_sorted
+            ]
         )
 
-        input_parameters_body = []
-        for module in session_modules:
-            input_parameters_body += [
-                line.lstrip(" ").rstrip(",")
-                for line in module["session_input_parameters_body"].split("\n")
-                if len(line.lstrip(" ").rstrip(",")) > 0
+        module_function_body = "\n".join(
+            [
+                f"{indentation_block}{return_dict_name}.update({sa.get_session_function_callblock()})"
+                for sa in session_artifacts_sorted
             ]
-
-        module_input_parameters_body = (
-            ",".join(input_parameters_body)
-            if len(input_parameters_body) <= 1
-            else "\n"
-            + f",\n{indentation_block}".join(input_parameters_body)
-            + ",\n"
         )
 
+        module_input_parameters = []
+        for sa in session_artifacts_sorted:
+            module_input_parameters += list(
+                sa.get_session_input_parameters_spec().values()
+            )
+        if len(module_input_parameters) == 0:
+            module_input_parameters_body = ""
+            parser_input_parameters = ""
+        elif len(module_input_parameters) == 1:
+            module_input_parameters_body = module_input_parameters[
+                0
+            ].default_args
+            parser_input_parameters = module_input_parameters[0].parser_args
```

Comment:
do we not need indentation_block for length == 1?

Reply:
see next one

Reply:
get rid of this part